yargs.js

```diff
@@ -172,56 +172,129 @@ function Yargs (processArgs, cwd, parentRequire) {
     frozen = undefined
   }
 
-  self.boolean = function (bools) {
-    options.boolean.push.apply(options.boolean, [].concat(bools))
+  self.boolean = function (keys) {
+    populateParserHintArray('boolean', keys)
     return self
   }
 
-  self.array = function (arrays) {
-    options.array.push.apply(options.array, [].concat(arrays))
+  self.array = function (keys) {
+    populateParserHintArray('array', keys)
     return self
   }
 
-  self.nargs = function (key, n) {
-    if (typeof key === 'object') {
-      Object.keys(key).forEach(function (k) {
-        self.nargs(k, key[k])
-      })
-    } else {
-      options.narg[key] = n
-    }
+  self.number = function (keys) {
+    populateParserHintArray('number', keys)
     return self
   }
 
-  self.number = function (numbers) {
-    options.number.push.apply(options.number, [].concat(numbers))
+  self.normalize = function (keys) {
+    populateParserHintArray('normalize', keys)
     return self
   }
 
-  self.choices = function (key, values) {
-    if (typeof key === 'object') {
-      Object.keys(key).forEach(function (k) {
-        self.choices(k, key[k])
-      })
-    } else {
-      options.choices[key] = (options.choices[key] || []).concat(values)
+  self.count = function (keys) {
+    populateParserHintArray('count', keys)
+    return self
+  }
+
+  self.string = function (keys) {
+    populateParserHintArray('string', keys)
+    return self
+  }
+
+  self.requiresArg = function (keys) {
+    populateParserHintArray('requiresArg', keys)
+    return self
+  }
+
+  self.skipValidation = function (keys) {
+    populateParserHintArray('skipValidation', keys)
+    return self
+  }
+
+  function populateParserHintArray (type, keys, value) {
+    keys = [].concat(keys)
+    keys.forEach(function (key) {
+      self.global(key)
```

Comment:
Instead of explicitly marking everything as global, I kinda figured it would be easier to just invert the reset logic to assume that everything is global unless explicitly marked as _non_-global.

In that approach, we could perhaps change `options.global` to be `options.local`, treat `yargs.global(opts, false)` as a synonym for a new API method `yargs.local(opts)`, and only worry about resetting local options in the reset method. (I think positional args for a command would be the only options to default as local.)

Thoughts? Is my view on the matter overly simplistic?